ui/src/attestation/contract.ts

```diff
@@ -0,0 +1,110 @@
+  claims: {
+    local: "0x785e8de68df899d77ce689f863e4166849c8bfd5",
+    ropsten: "0xF8F22AA794DDA79aC0C634a381De0226f369bCCe",
+  },
+};
+
+// Get the address of the Claims Contract for the given environment
+export function claimsAddress(environment: ethereum.Environment): string {
+  switch (environment) {
+    case ethereum.Environment.Local:
+      return addresses.claims.local;
+    case ethereum.Environment.Ropsten:
+      return addresses.claims.ropsten;
+  }
+}
+
+const FORMAT_SHA1 = ethers.BigNumber.from(1);
+const FORMAT_SHA1_LENGTH = 20;
+
+export function claims(signer: Signer, address: string): ClaimsContract {
+  return new ClaimsContract(signer, address);
+}
+
+export class ClaimsContract {
+  contract: Claims;
+
+  constructor(signer: Signer, address: string) {
+    this.contract = ClaimsFactory.connect(address, signer);
+  }
+
+  async claim(urn: string): Promise<void> {
+    const payload = parseIdentitySha1(urn);
+    const tx = await this.contract.claim(FORMAT_SHA1, payload);
+    transaction.add(transaction.claimRadicleIdentity(tx, urn));
+  }
+
+  // Start watching claims of a given Ethereum address.
+  // `onClaimed` is called immediately with the latest claim or `undefined` if there was none.
+  // Returns a function, which unwatches claims when called.
+  async watchClaimed(
+    address: string,
+    onClaimed: (claimed?: Uint8Array) => void
+  ): Promise<() => void> {
+    const filter = this.contract.filters.Claimed(address);
+
+    await this.contract.on(filter, async (_: unknown, event: ethers.Event) => {
+      const claimed = await this.getClaimed(event.transactionHash, address);
+      onClaimed(claimed);
+    });
+
+    const lastEvent = (await this.contract.queryFilter(filter)).pop();
+    if (lastEvent !== undefined) {
+      const lastClaimed = await this.getClaimed(
+        lastEvent.transactionHash,
+        address
+      );
+      onClaimed(lastClaimed);
+    } else {
+      onClaimed(undefined);
+    }
+    return () => {
+      this.contract.off(filter, onClaimed);
+    };
+  }
+
+  // Extracts the claimed identity root from the transaction sent by the address
+  private async getClaimed(
+    txHash: string,
+    address: string
+  ): Promise<Uint8Array> {
+    const tx = await this.contract.provider.getTransaction(txHash);
+    if (tx === null) {
+      throw new Error("Claim transaction not found");
+    }
+    // TODO(igor) add more checks for malicious client (e.g. TX signature, network, etc.)
+    if (tx.from.toLowerCase() != address.toLowerCase()) {
+      throw new Error("Claim transaction sent by an invalid address");
+    }
+    if (tx.to.toLowerCase() != this.contract.address.toLowerCase()) {
+      throw new Error("Claim transaction sent to an invalid contract");
+    }
+    const [format, payloadRaw]: [
+      ethers.BigNumberish,
+      ethers.BytesLike
+    ] = this.contract.interface.decodeFunctionData("claim", tx.data);
+    if (FORMAT_SHA1.eq(format) == false) {
```

Comment:
Unless there's a good reason not to, we should stick to the convention of only using strict equality operators: `!==` and `===`. Noticed also in a couple of other places in this changeset.

From [JS the good parts]( https://www.goodreads.com/book/show/2998152-javascript):
> JavaScript has two sets of equality operators: === and !==, and their evil twins == and !=. The good ones work the way you would expect. If the two operands are of the same type and have the same value, then === produces true and !== produces false. The evil twins do the right thing when the operands are of the same type, but if they are of different types, they attempt to coerce the values. the rules by which they do that are complicated and unmemorable.

Reply:
Fixed